cli/lsp/language_server.rs

```diff
@@ -1493,22 +1493,30 @@ impl Inner {
       .collect();
 
     // if the current deno.json has changed, we need to reload it
-    if let Some(config_file) = self.config.maybe_config_file() {
-      if changes.contains(&config_file.specifier)
-        || self
-          .config
-          .maybe_lockfile()
-          .map(|l| has_lockfile_changed(&l.lock(), &changes))
-          .unwrap_or(false)
-      {
-        if let Err(err) = self.update_config_file().await {
-          self.client.show_message(MessageType::WARNING, err);
-        }
-        if let Err(err) = self.update_tsconfig().await {
-          self.client.show_message(MessageType::WARNING, err);
-        }
-        touched = true;
+    let has_config_changed = match self.config.maybe_config_file() {
+      Some(config_file) => changes.contains(&config_file.specifier),
+      None => {
+        // check for auto-discovery
+        changes.iter().any(|url| {
+          url.path().ends_with("/deno.json")
+            || url.path().ends_with("/deno.jsonc")
```

Comment:
Won't the slash cause problems on Windows?

Reply:
They're urls :)